Source/Train-Railway/Program.cs

```diff
@@ -0,0 +1,312 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Threading;
+
+
+namespace Train_Railway
+{
+    class Program
+    {
+        static List<Timetable> timeTables = new List<Timetable>();
+        static List<Passenger> passengers = new List<Passenger>();
+        static List<Station> stations = new List<Station>();
+        static List<Train> trains = new List<Train>();
+        static List<Track> tracks = new List<Track>();
+
+        //static FilesPath m = new FilesPath();
+        static Clock clock = new Clock(10, 19);
+        static object stationLock = new object();
+
+        static void Main(string[] args)
+        {
+            IninitalizeData();
+
+            Thread track1 = new Thread(StartTrain1);
+            Thread track2 = new Thread(StartTrain2);
+            track1.Start();
```

Comment:
Vi kanske förstår koden fel nu, men som vi tänker så ser det ut som att ni startar tråden och sedan läser in all data istället för tvärtom. Det kanske hade varit lättare att läsa in datan först och sedan skapa tråden. Nu blir det att tråden behöver gör uträkningar för att hämta sin data , vilket känns som att man belastar tråden på ett onödigt sätt istället för att bara ge den all information som den behöver direkt och så förhåller sig den till det. Men vi kan ha förståt det fel med era startTrain metoder.